server/py/services/api/crud/model_monitoring/deployment.py

```diff
@@ -1440,13 +1486,22 @@ def _extract_model_endpoints_from_function_graph(
             mlrun.serving.states.RouterStep, mlrun.serving.states.RootFlowStep
         ],
         sampling_percentage: float,
-    ) -> list[
-        tuple[
-            mlrun.common.schemas.ModelEndpoint,
-            mm_constants.ModelEndpointCreationStrategy,
-            str,
-        ]
+        model_endpoints: list[ModelEndpoint],
+    ) -> tuple[
+        list[
+            tuple[
+                mlrun.common.schemas.ModelEndpoint,
+                mm_constants.ModelEndpointCreationStrategy,
+                str,
+            ]
+        ],
+        typing.Union[
+            mlrun.serving.states.RouterStep, mlrun.serving.states.RootFlowStep
+        ],
     ]:
+        endpoints_names_uid_map = {
+            endpoint.model_name: endpoint.uid for endpoint in model_endpoints
+        }
```

Comment:
You should do as part of the list db function. Only in there the key needs to be - "<project_name>-<function_name>-<function_tag>-<name>".